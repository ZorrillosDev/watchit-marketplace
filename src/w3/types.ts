
export interface NetworkSetting {
  PURCHASE_GATEWAY: string,
  NFT: string,
  WVC: string
}
