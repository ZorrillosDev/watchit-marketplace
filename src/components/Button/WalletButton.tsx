import React, { FC } from 'react'
import RoundButton from '@components/Button/RoundButton'
import AccountBalanceWalletIcon from '@material-ui/icons/AccountBalanceWallet'
import i18n from '@src/i18n'

const WalletButton: FC = (): JSX.Element => {
  return (
    <RoundButton
      variant='contained' color='primary'
      startIcon={<AccountBalanceWalletIcon fontSize='inherit' />}
    >
      {i18n.t('GLOBAL_WALLET')}
    </RoundButton>
  )
}

export default WalletButton
