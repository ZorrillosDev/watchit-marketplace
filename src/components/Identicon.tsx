import React, {FC, useMemo} from 'react'

import {createAvatar} from '@dicebear/avatars'
import * as style from '@dicebear/adventurer-neutral'
import {String} from '@src/utils'

interface IdenticonProps {
    seed: string
    size?: number
    radius?: number
    backgroundColor?: string
}

export const PixelArtIdenticon: FC<IdenticonProps> = (props): JSX.Element => {
    const randomHex = useMemo(() => String.toHex(props.seed), [props.seed])
    const {seed = '', size = 25, radius = 30, backgroundColor = `${randomHex}`} = props

    return (
        <div
            style={{width: size, height: size}}
            dangerouslySetInnerHTML={{
                __html: createAvatar(style, {
                    seed, size, radius, backgroundColor
                })
            }}
        />
    )
}
