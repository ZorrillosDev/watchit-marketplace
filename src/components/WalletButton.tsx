import { useTranslation } from 'react-i18next'
import { useWeb3React } from '@web3-react/core'
import { RoundButton } from '@components/index'
import React, { FunctionComponent, ReactElement } from 'react'
import { connect, RootStateOrAny } from 'react-redux'
import AccountBalanceWalletIcon from '@material-ui/icons/AccountBalanceWallet'
import { useMetamaskListener, useMetamaskEager } from '@hooks/WalletHooks'
import * as walletActions from '@state/actions/WalletActions'
import { Injected } from '@src/web3'

const WalletButton: FunctionComponent = ({ wallet, connect, connected }: any): ReactElement => {
  const { t } = useTranslation()
  const { activate } = useWeb3React()

  // handle logic to eagerly connect to the injected ethereum provider, if it exists and has granted access already
  const triedEager = useMetamaskEager()
  const requestConnection = !triedEager || !!wallet.connecting

  // handle logic to connect in reaction to certain events on the injected ethereum provider, if it exists
  useMetamaskListener(requestConnection)

  return (
    <RoundButton
      variant='contained' color='secondary'
      startIcon={<AccountBalanceWalletIcon fontSize='inherit' />}
      onClick={async () => {
        connect()
        await activate(Injected)
      }}
    >
      {wallet.connecting ? t('GLOBAL_CONNECTING') : t('GLOBAL_WALLET')}
    </RoundButton>
  )
}

const mapDispatchToProps = {
  connect: walletActions.connect,
  connected: walletActions.connected,
  error: walletActions.error
}

const mapStateToProps = (state: RootStateOrAny): any => {
  return {
    wallet: state.wallet
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(WalletButton)
