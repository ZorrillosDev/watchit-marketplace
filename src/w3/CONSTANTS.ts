import WNFT from '@abi/WNFT.json'

export const RINKEBY_ALCHEMY_API_KEY = process.env.RINKEBY_ALCHEMY_API_KEY ?? ''
export const BLACK_HOLE = '0x0000000000000000000000000000000000000000'
export { WNFT }
