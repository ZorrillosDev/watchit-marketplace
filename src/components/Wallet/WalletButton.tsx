import React, { FC, memo } from 'react';
import i18n from '@src/i18n';
import { Button } from '@mui/material';
import { PixelArtIdenticon } from '@components/Identicon';
import { AccountBalanceWallet } from '@components/Icons';
import { Ethers } from '@src/utils';
import { BigNumber } from '@ethersproject/bignumber';

// ===========================|| WALLET - BUTTON ||=========================== //

/* eslint-disable  react/display-name */

export interface WalletButtonProps {
  etherBalance?: BigNumber
  account?: string | null
  activateBrowserWallet: () => void
  onClick: () => void
}

export const WalletButton: FC<WalletButtonProps> = (props): JSX.Element => {
  let { account, activateBrowserWallet, etherBalance } = props;

  const handleClick = () => {
    if (account === undefined) activateBrowserWallet();
    props.onClick();
  };

  const icon: JSX.Element = account !== undefined
    ? <PixelArtIdenticon seed={account ?? ''} />
    : <AccountBalanceWallet fontSize='inherit' />;

  return (
    <Button
      sx={{
        mt: { xs: 1, md: 0 },
        ml: { xs: 0, md: 1 },
        borderRadius: 3,
      }}
      variant='contained'
      color='primary'
      startIcon={icon}
      onClick={handleClick}
    >
      {
        account !== undefined
          ? `${(etherBalance != null) ? Ethers.getWeiToETH(etherBalance) : 0} ETH`
          : i18n.t('GLOBAL_WALLET')
      }
    </Button>
  );
};

export default memo(WalletButton);
