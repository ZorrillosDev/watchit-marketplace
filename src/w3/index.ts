import { ChainId, Config } from '@usedapp/core'
import { NetworkSetting } from '@w3/types'
import { RINKEBY_ALCHEMY_API_KEY } from '@w3/CONSTANTS'

export const config: Config = {
  readOnlyChainId: ChainId.Rinkeby,
  readOnlyUrls: {
    [ChainId.Rinkeby]: `https://eth-rinkeby.alchemyapi.io/v2/${RINKEBY_ALCHEMY_API_KEY}`
  }
}

export function getNetworkSettings (networkName: ChainId | undefined): NetworkSetting {
  /**
     * @param {ChainId} networkName network to retrieve contract
     * @return {NetworkSetting}
     */

  const contractAddressCollection: { [key: number]: NetworkSetting } = {
    [ChainId.Kovan]: {
      PURCHASE_GATEWAY: process.env.KOVAN_CONTRACT_PURCHASE_GATEWAY ?? '',
      NFT: process.env.KOVAN_CONTRACT_NFT ?? '',
      WVC: process.env.KOVAN_CONTRACT_FT ?? ''
    },
    [ChainId.Rinkeby]: {
      PURCHASE_GATEWAY: process.env.RINKEBY_CONTRACT_PURCHASE_GATEWAY ?? '',
      NFT: process.env.RINKEBY_CONTRACT_NFT ?? '',
      WVC: process.env.RINKEBY_CONTRACT_FT ?? ''

    }
  }

  // return default rikeby if not supported chain
  if (networkName === undefined || !(networkName in contractAddressCollection))
    return contractAddressCollection[ChainId.Rinkeby]

  return contractAddressCollection[networkName]
}
