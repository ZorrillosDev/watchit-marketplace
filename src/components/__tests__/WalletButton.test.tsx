import React from 'react';
import { shallow, mount } from 'enzyme';
import WalletButton from '@components/WalletButton';
import * as web3Core from '@usedapp/core';
import 'jest-styled-components';
import { Button } from '@mui/material';
import { DAppProvider } from '@usedapp/core';
import { AccountBalanceWallet } from '@mui/icons-material';
import { PixelArtIdenticon } from '@components/Identicon';
import { config } from '@src/w3/';

jest.mock('@usedapp/core');

const useEthersType = {
  active: true,
  connector: undefined,
  error: undefined,
  library: undefined,
  chainId: undefined,
  account: '0x0',
  setError: jest.fn(),
  deactivate: jest.fn(),
  activateBrowserWallet: jest.fn(),
  activate: jest.fn(),
  isLoading: false,
  switchNetwork: jest.fn(),
};

/* eslint-disable no-undef */
describe('<WalletButton/>', () => {
  beforeEach(() => {
    jest.spyOn(web3Core, 'useEthers')
      .mockImplementation(() => ({ ...useEthersType }));
  });

  it('should render', () => {
    const walletButtonComponent = shallow(
      <DAppProvider config={config}>
        <WalletButton />
      </DAppProvider>,
    );
    expect(walletButtonComponent).toMatchSnapshot();
  });

  it('should display english translation', () => {
    jest.spyOn(web3Core, 'useEthers')
      .mockImplementation(() => ({ ...useEthersType, account: undefined }));

    const walletButtonComponent = mount(<WalletButton />);
    const button = walletButtonComponent.find(Button);
    expect(walletButtonComponent.find(AccountBalanceWallet).length).toBe(1);
    expect(button.text()).toBe('Connect wallet');
  });

  it('should display wallet connection if success with metamask', () => {
    jest.spyOn(web3Core, 'useEthers')
      .mockImplementation(() => ({ ...useEthersType }));

    const walletButtonComponent = mount(<WalletButton />);
    expect(walletButtonComponent.find(PixelArtIdenticon).length).toBe(1);
  });
});


