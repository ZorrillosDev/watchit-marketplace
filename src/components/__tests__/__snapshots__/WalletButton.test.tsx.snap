// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletButton/> should render 1`] = `ShallowWrapper {}`;
